### PR TITLE
Fix client ae response logic to use knowledge

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

**Comment by joshuef** [comment]
bors try

File: sn_client/src/connections/listeners.rs [hunk 1]
```diff
@@ -279,14 +280,22 @@ impl Session {
             "Bounced msg {msg_id:?} received in an AE response: {client_msg:?} from {src_peer:?}"
         );
 
-        let target_elders: Vec<_> = received_sap.elders_vec();
+        let knowlege = self.network.read().await;
+        // Get the best sap we know of now.
+        // We don't just rely on the returned SAP, as we should be updating the knowledge if it's valid, before we get here.
+        let best_sap = match knowlege.closest(&bounced_msg_dst.name, None) {
+            Some(sap) => sap,
+            None => return Err(Error::NoCloseSapFound),
```
Comment (bochaco) [inline sn_client/src/connections/listeners.rs:288]: This more succint way is perhaps better here: `let best_sap = knowlege.closest(&bounced_msg_dst.name, None).ok_or(Error::NoCloseSapFound)?;`

File: sn_client/src/errors.rs [hunk 2]
```diff
@@ -29,6 +29,10 @@ pub type Result<T, E = Error> = std::result::Result<T, E>;
 #[derive(Error, Debug)]
 #[non_exhaustive]
 pub enum Error {
+    /// No section was found to be closest for given dataname/
+    /// This is indicative of some larger problem
+    #[error("No SAP for data name")]
+    NoCloseSapFound,
```
Comment (bochaco) [inline sn_client/src/errors.rs:35]: Perhaps adding context info will be helpfull, i.e. the xorname no close SAP was found for

**Review by bochaco — APPROVED** [review]

**Pull request merged**